### PR TITLE
Test some queries aimed at identifying common testthat calls

### PR DESCRIPTION
**Pull request by jennybc**
As discussed elsewhere, I'm adding tests for specific queries related to testthat usage. I've made a lot of choices, e.g. around helpers, that I'm happy to discuss and tweak. Having these queries under test is useful to me and I feel like it delivers some value for this package and the associated R grammar, more in the spirit of integration tests, I suppose.

The high level goal is to identify individual test atoms (meaning calls to `test_that()`, `describe()`, and `it()`, with their associated `desc` or `description`) and to get information on the tree, (meaning the parent `describe()` for each `it()` and whether a specific call is at the top-level or nested within another call). A very relevant testthat feature is the ability to run an individual `test_that()` or `describe()` via `testthat::test_file(desc =)`.

The query for `testthat::test_that()` is more ambitious, in terms of what it includes and excludes. The query for `describe(it())` is less so. This is partially because even a minimal `describe(it())` query has to be more complicated, by necessity. Also, I believe `test_that()` has vastly more usage, which makes it seem more worthwhile to contemplate a larger set of possibilities.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -16,6 +16,7 @@ Depends:
     R (>= 4.0.0)
 Imports:
     cli (>= 3.6.2),
+    glue,
```
Comment (jennybc) [inline DESCRIPTION]: I feel like glue is useful when preparing the little code snippets, we usually regard it as "free", and it's already a hard indirect dependency.

``` r
pak::pkg_deps_explain("DavisVaughan/r-tree-sitter", "glue")
#> ℹ Loading metadata database
#> ✔ Loading metadata database ... done
#> 
#> treesitter -> vctrs -> glue
#> treesitter -> vctrs -> lifecycle -> glue
```

File: tests/testthat/fixtures/describe.scm [hunk 2]
```diff
@@ -0,0 +1,32 @@
+(call
+	function: [
+		(identifier) @parent_function
+		(namespace_operator
+			lhs: (identifier) @pkg (#eq? @pkg "testthat")
```
Comment (jennybc) [inline tests/testthat/fixtures/describe.scm]: Thanks for the discussion that lead to me learning these predicates could go here, inside the alternation!

File: tests/testthat/helper-describe.R [hunk 3]
```diff
@@ -0,0 +1,30 @@
+expect_describe_it_captures <-function(code_source, parent_desc, desc, captures) {
```
Comment (jennybc) [inline tests/testthat/helper-describe.R]: Not entirely sure where to put this, but without such a helper, the tests were extremely long and repetitive. This is highly specific to the `.scm` files I've included, obviously.

File: tests/testthat/helper.R [hunk 4]
```diff
@@ -0,0 +1,28 @@
+read_file <- function(file) {
```
Comment (jennybc) [inline tests/testthat/helper.R]: Again, happy to discuss the helpers -- which ones to define and where to do so.

File: tests/testthat/test-query-describe.R [hunk 5]
```diff
@@ -0,0 +1,84 @@
+test_that("describe(it()) with 2 it()s can be matched", {
```
Comment (jennybc) [inline tests/testthat/test-query-describe.R:1]: At a very high-level, the tests for `describe(it())` and `test_that()` come from a collection of informal tests I've been using to develop queries in the tree-sitter playground.

For `describe(it())`, I'm just drawing on the official examples from testthat.

File: tests/testthat/test-query-describe.R [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  # first example for testthat::describe()
+  code_source <- glue::glue('
```
Comment (jennybc) [inline tests/testthat/test-query-describe.R]: I like how glue lets you lay this out with the correct internal indentation (internal to the specimen) and the indentation related to being inside `test_that()` gets removed.

**Comment by DavisVaughan** [comment]
Closing in favor of https://github.com/r-lib/tree-sitter-r/pull/106, which runs every time we touch the grammar, so we can ensure we never break these on accident

**Pull request closed without merging**